Sources/WalletConnectAuth/Auth/Auth.swift

```diff
@@ -0,0 +1,251 @@
+
+import Foundation
+import WalletConnectUtils
+import WalletConnectRelay
+import Combine
+
+public class Auth {
+    static public let instance = Auth()
+    private let client: AuthClient
+    private static var config: Config?
+    public let logger: ConsoleLogging
+    private let relayClient: RelayClient
+    
+    private init() {
+        guard let config = Auth.config else {
+            fatalError("Error - you must configure before accessing Auth.instance")
+        }
+        relayClient = RelayClient(relayHost: "relay.walletconnect.com", projectId: config.projectId, socketConnectionType: config.socketConnectionType)
+        client = AuthClient(metadata: config.metadata, relayClient: relayClient)
+        self.logger = client.logger
```

Comment:
Could we avoid this inverted proxy somehow?

Reply:
do you mean AuthClient in Auth?